protocol/shannon/websocket_context.go

```diff
@@ -0,0 +1,227 @@
+package shannon
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/pokt-network/poktroll/pkg/relayer/proxy"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	sdk "github.com/pokt-network/shannon-sdk"
+
+	"github.com/buildwithgrove/path/gateway"
+	protocolobservations "github.com/buildwithgrove/path/observation/protocol"
+	"github.com/buildwithgrove/path/request"
+)
+
+// The requestContext implements the gateway.ProtocolRequestContext interface.
+// It handles protocol-level WebSocket message processing for both client and endpoint messages.
+// For example, client messages are signed and endpoint messages are validated.
+var _ gateway.ProtocolRequestContext = &requestContext{}
```

Comment:
Is this the right place for it?

Reply:
I've moved all methods to a new `ProtocolRequestContextWebsocket` interface for clarity, including updating this check.

This said, we may still want to revisit the architecture but I believe that for the time being this is sufficient.